ring-core/test/ring/middleware/multipart_params/test/multipart_params.clj

```diff
@@ -0,0 +1,27 @@
+(ns ring.middleware.multipart-params.test.multipart-params
+  (:require [clojure.test :refer :all]
+            [ring.middleware.multipart-params :refer :all]
+            [ring.util.response :as response])
+  (:import org.apache.commons.fileupload.InvalidFileNameException))
+
+(deftest wrap-multipart-params-error-test
+  (let [invalid-file-name "foo\\0.bar"
+        error-msg (str "Invalid file name: " invalid-file-name)]
+    (with-redefs [multipart-params-request
+                  (fn [& _]
+                    (throw (InvalidFileNameException.
+                            invalid-file-name
+                            error-msg)))]
+      (testing "Synchronous error response"
+        (let [handler (wrap-multipart-params (constantly (response/response "Ok")))]
+          (is (= (response/bad-request error-msg)
```

Comment:
Prefer using the full request map in tests, over helper functions like `bad-request`.